### PR TITLE
KTOR-1114 Core Build

### PR DESCRIPTION
**Pull request by hhariri**
Clean up core files and set correct tasks for gradle

### CODE REVIEW

File: .teamcity/src/VcsRoots.kt [hunk 1]
```diff
@@ -1,24 +1,42 @@
+package subprojects
```
Comment (Stexxe) [inline .teamcity/src/VcsRoots.kt:1]: Maybe we should move the file `VcsRoots.kt` to the `subprojects` directory

File: .teamcity/src/VcsRoots.kt [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
+    vcs {
+        root(VCSCore)
+    }
+    triggers {
+        vcs {
+            quietPeriodMode = VcsTrigger.QuietPeriodMode.USE_DEFAULT
+        }
+    }
+}
+
```
Comment (Stexxe) [inline .teamcity/src/VcsRoots.kt:40]: Empty lines

File: .teamcity/src/subprojects/build/core/NativeBuild.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+class NativeBuild(private val osEntry: OSEntry) : BuildType({
+    id("KtorMatrixNative_${osEntry.name}".toExtId())
+    name = "Native on ${osEntry.name}"
+
+    setupDefaultVCSRootAndTriggers()
+
+    steps {
+        gradle {
+            name = "Build and Run Tests"
+            tasks = "ktor-client:ktor-client-curl:${osEntry.taskName} --info"
```
Comment (Stexxe) [inline .teamcity/src/subprojects/build/core/NativeBuild.kt:17]: Is it enough to just build `ktor-client-curl` for the native?

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: .teamcity/src/subprojects/build/core/NativeBuild.kt (hunk 3 above)
Comment (hhariri) [reply .teamcity/src/subprojects/build/core/NativeBuild.kt:17]: I believe so, yes

File: .teamcity/src/VcsRoots.kt (hunk 1 above)
Comment (hhariri) [reply .teamcity/src/VcsRoots.kt:1]: Thought it was. My bad.

**Review by Stexxe — APPROVED** [review]

**Pull request merged**